### DISCUSSION TITLE
Validation blocks rendering

### DISCUSSION THREAD

**Discussion by codingcircus** [react-hook-form/react-hook-form#4072] [Q&A]
Hi. I have a fairly complex form, where one of the inputs affects changes which input fields are displayed below it. In general validating this form via yup and react-hook-form works pretty well, but when the particular controling input changes, it takes quite long (~ 500 - 600 ms) for the form to rerender. I noticed now, that when I disable the yup resolver, the rendering updates happen almost instantly (~ 100ms). Is there a clean way to disable or offset the validation during this rerender?

**Comment by bluebill1049** [comment]
you can build a custom resolver, and leverage the context object to control when you want to trigger the validation.

**Reply by codingcircus** [reply]
I basically solved it now, by wrapping the yupResolver in a custom resolver:
```
import {yupResolver} from '@hookform/resolvers';

export default (schema, disabled) => {
  if (!disabled) {
    return async values => ({values, errors: {}});
  }

  return yupResolver(schema);
};
```
is this roughly what you meant?

**Reply by bluebill1049** [reply]
yes, but i prefer keep context as an object.
